pygeoapi/provider/postgresql.py

```diff
@@ -0,0 +1,295 @@
+                str(self.conn_dic)))
+            raise ProviderConnectionError()
+
+        self.cur = self.conn.cursor()
+        if self.context == 'query':
+            # Getting columns
+            query_cols = "SELECT column_name FROM information_schema.columns \
+            WHERE table_name = '{}' and udt_name != 'geometry';".format(
+                self.table)
+
+            self.cur.execute(query_cols)
+            result = self.cur.fetchall()
+            self.columns = SQL(', ').join(
+                [Identifier(item[0]) for item in result]
+                )
+
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        # some logic to commit/rollback
+        self.conn.close()
+
+
+class PostgreSQLProvider(BaseProvider):
+    """Generic provider for Postgresql based on psycopg2
+    using sync approach and server side
+    cursor (using support class DatabaseCursor)
+    """
+
+    def __init__(self, provider_def):
+        """
+        PostgreSQLProvider Class constructor
+
+        :param provider_def: provider definitions from yml pygeoapi-config.
+                             data,id_field, name set in parent class
+                             data contains the connection information
+                             for class DatabaseCursor
+
+        :returns: pygeoapi.providers.base.PostgreSQLProvider
+        """
+
+        BaseProvider.__init__(self, provider_def)
+
+        self.table = provider_def['table']
+        self.id_field = provider_def['id_field']
+        self.conn_dic = provider_def['data']
+
+        LOGGER.debug('Setting Postgresql properties:')
+        LOGGER.debug('Connection String:{}'.format(
+            ",".join(("{}={}".format(*i) for i in self.conn_dic.items()))))
+        LOGGER.debug('Name:{}'.format(self.name))
+        LOGGER.debug('ID_field:{}'.format(self.id_field))
+        LOGGER.debug('Table:{}'.format(self.table))
+
+    def query(self, startindex=0, limit=10, resulttype='results',
+              bbox=[], time=None, properties=[]):
+        """
+        Query Postgis for all the content.
+        e,g: http://localhost:5000/collections/hotosm_bdi_waterways/items?
+        limit=1&resulttype=results
+
+        :param startindex: starting record to return (default 0)
+        :param limit: number of records to return (default 10)
+        :param resulttype: return results or hit limit (default results)
+        :param bbox: bounding box [minx,miny,maxx,maxy]
+        :param time: temporal (datestamp or extent)
+        :param properties: list of tuples (name, value)
+
+        :returns: GeoJSON FeaturesCollection
+        """
+        LOGGER.debug('Querying PostGIS')
+
+        if resulttype == 'hits':
+
+            with DatabaseConnection(self.conn_dic,
+                                    self.table, context="hits") as db:
+                cursor = db.conn.cursor(cursor_factory=RealDictCursor)
+                sql_query = SQL("select count(*) as hits from {}").\
```

Comment:
Can we implement binding here to make safer against SQL injections?

Reply:
Identifier(self.table) on line 182 should take care of the SQL injection as indicated in http://initd.org/psycopg/docs/sql.html  as explained on class psycopg2.sql.SQL(string) and on  class psycopg2.sql.Identifier(*strings)